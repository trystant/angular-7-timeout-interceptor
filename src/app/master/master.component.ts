import { Component } from '@angular/core';

@Component({
  template: `
    <h1>Angular with Jest sauce</h1>

    <ul>
      <li><a routerLink="/todo" routerLinkActive="active">Todo sub-app</a></li>
    </ul>

    <router-outlet></router-outlet>
  `
})
export class MasterComponent {
}
