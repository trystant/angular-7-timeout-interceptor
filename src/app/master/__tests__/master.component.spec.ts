import { ConfigureFn, configureTests } from '@lib/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { MasterComponent } from '../master.component';
import { ComponentFixture } from '@angular/core/testing';

describe('MasterComponent', () => {
  let fixture: ComponentFixture<MasterComponent>;
  let component: MasterComponent;

  beforeEach(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [MasterComponent],
        imports: [RouterTestingModule, NoopAnimationsModule],
        schemas: [NO_ERRORS_SCHEMA]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(MasterComponent);
      component = fixture.componentInstance;
      fixture.detectChanges();
    });
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should render title in a h1 tag', () => {
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('Angular with Jest sauce');
  });
});
