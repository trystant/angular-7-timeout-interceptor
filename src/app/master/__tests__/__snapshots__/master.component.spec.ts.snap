// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterComponent should match snapshot 1`] = `
<ng-component>
  <h1>
    Angular with Jest sauce
  </h1>
  <ul>
    <li>
      <a
        click={[Function]}
        href="/todo"
        ng-reflect-router-link="/todo"
        ng-reflect-router-link-active="active"
        routerLink="/todo"
        routerLinkActive="active"
        target={null}
      >
        Todo sub-app
      </a>
    </li>
  </ul>
  <router-outlet />
</ng-component>
`;
